exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/UncheckedFlatMapProof.java

```diff
@@ -0,0 +1,176 @@
+package com.exonum.binding.storage.proofs.map.flat;
+
+import static com.exonum.binding.hash.Funnels.hashCodeFunnel;
+import static com.exonum.binding.storage.proofs.DbKeyFunnel.dbKeyFunnel;
+
+import com.exonum.binding.hash.HashCode;
+import com.exonum.binding.hash.HashFunction;
+import com.exonum.binding.hash.Hashing;
+import com.exonum.binding.storage.proofs.map.DbKey;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * An unchecked flat map proof, which does not include any intermediate nodes.
+ */
+public class UncheckedFlatMapProof implements UncheckedMapProof {
+
+  private static final HashFunction HASH_FUNCTION = Hashing.defaultHashFunction();
+
+  private final List<MapProofEntry> proofList;
+
+  private final Set<byte[]> requestedKeys;
+
+  UncheckedFlatMapProof(List<MapProofEntry> proofList, Set<byte[]> requestedKeys) {
+    this.proofList = proofList;
+    this.requestedKeys = requestedKeys;
+  }
+
+  @SuppressWarnings("unused") // Native API
+  static UncheckedFlatMapProof fromUnsorted(MapProofEntry[] proofList, byte[][] requestedKeys) {
+    List<MapProofEntry> mapProofEntries = Arrays.asList(proofList);
+    mapProofEntries.sort(Comparator.comparing(MapProofEntry::getDbKey));
+    Set<byte[]> expectedKeys = new HashSet<>(Arrays.asList(requestedKeys));
```

Comment:
I’ve done the same mistake — an array does not override equals and hashCode, therefore, cannot be reliably used in a _hash_ set. If we need to have a set of byte[], then we have to add a ByteString class or use a specialized set (e.g., a TreeSet + a comparator).

I’d consider adding a ByteString interface with the minimum required methods and static factory methods + ArrayByteString implementing that interface — this way it will be easier to replace it if needed.

Some existing implementations (overkill for our needs at the moment, just for reference): 
* https://github.com/google/protobuf/blob/master/java/core/src/main/java/com/google/protobuf/ByteString.java or
* https://github.com/kbolino/ByteString/blob/master/src/main/java/com/kbolino/libraries/bytestring/ByteString.java